pkg/reference/ref.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reference
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+var (
+	// DefaultParentKind is the default metav1.GroupKind for ParentReferences.
+	DefaultParentKind = metav1.GroupKind{
+		Group: v1alpha2.GroupName,
+		Kind:  "Gateway",
+	}
+
+	// DefaultSecretKind is the default metav1.GroupKind for SecretObjectReferences.
+	DefaultSecretKind = metav1.GroupKind{
+		Group: "",
+		Kind:  "Secret",
+	}
+
+	// DefaultBackendObjectKind is the default metav1.GroupKind for BackendObjectReferences.
+	DefaultBackendObjectKind = metav1.GroupKind{
```

Comment:
Why not DefaultBackendKind or DefaultServiceKind?

Reply:
These are default for the type of ref. So BackendObjectReference -> reference.Backend -> reference.BackendObjectKind.

The kinds here are mainly for readability because CRD defaulting means we should always have group and kind values.